panels/models/attraction.py

```diff
@@ -33,6 +34,30 @@ def sluggify(s):
     return RE_SLUG.sub('-', s).lower().strip('-')
 
 
+def treeify(items, keys, value_key=None):
+    if not keys:
+        return items
+    keys = listify(keys)
+    last_key = keys[-1]
+    treeified = OrderedDict()
+    for item in items:
+        current = treeified
```

Comment:
Wait, are you.. transmuting the original OrderedDict() by assigning it to a second variable, which you then modify? If so, that really obfuscates what's happening. If not, I have no idea how `treeified` is ever modified.

Reply:
I'm making _nested_ OrderedDicts. So when we start the iteration, the initial OrderedDict is the current one. As we go deeper in the tree, `current` will point to child OrderedDicts.